avro/src/main/java/com/fasterxml/jackson/dataformat/avro/AvroAnnotationIntrospector.java

```diff
@@ -113,6 +124,10 @@ public Object findSerializer(Annotated a) {
         if (a.hasAnnotation(Stringable.class)) {
             return ToStringSerializer.class;
         }
+        AvroEncode ann = _findAnnotation(a, AvroEncode.class);
+        if (ann != null) {
+            return CustomEncodingSerializer.class;
```

Comment:
The `_isExplicitClassOrOb` check [here](https://github.com/FasterXML/jackson-databind/blob/master/src/main/java/com/fasterxml/jackson/databind/introspect/AnnotationIntrospectorPair.java#L315) requires that the primary return a `Class` object - If you return an instance of a `JsonSerializer`, `_isExplicitClassOrOb` returns false and the secondary is used. If that's unintended behavior, then yeah, that should be fixed, and this code could be simplified to just return a `JsonSerializer` instance with the context information already bound instead of deferring that to a custom `HandlerInstantiator` (and the same for `JsonDeserializer` above)